### PR TITLE
Update DML EP to accept broadcasted tensor of size 1 to match CPU

### PR DESCRIPTION
**Pull request by chrilaMSFT**
### Description
With QDQ enabled for Dml EP we are seeing some models not optimize constant nodes with incorrect tensor size of scale[1] and zeropoint[1] that does not match the input size.  CPU accepts this parameter type so updating Dml EP to match CPU behavior.

### Motivation and Context
Want to match CPU EP behavior.

### CODE REVIEW

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp [hunk 1]
```diff
@@ -558,7 +558,9 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
             {
                 ML_CHECK_VALID_ARGUMENT(axis < outputShapeDimCount);
                 uint32_t broadcastAxisLength = outputShape[axis];
-                ML_CHECK_VALID_ARGUMENT(inputTensorShape[0] == broadcastAxisLength);
+                ML_CHECK_VALID_ARGUMENT((inputTensorShape[0] == broadcastAxisLength) ||
+                                        // treat as broadcast dimension to match CPU behavior
+                                        (inputTensorShape[0] == 1));
```
Comment (fdwr) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp:563]: [nit] Consistent indentation with rest of DML EP (uniformly 4 spaces per indentation level):

```
                ML_CHECK_VALID_ARGUMENT(
                    (inputTensorShape[0] == broadcastAxisLength) ||
                    // treat as broadcast dimension to match CPU behavior
                    (inputTensorShape[0] == 1)
                ); 
```

**Review by fdwr — COMMENTED** [review]
I'm okay with it, even if it's technically incorrect [according to the spec](https://onnx.ai/onnx/operators/onnx__DequantizeLinear.html), because it's harmless, and evidently `com.microsoft.DequantizeLinear` differs from `ai.onnx.DequantizeLinear`.

**Comment by chrilaMSFT** [comment]
I think CPU and Dml should match behavior.  Also since it hit on one of the converted models I added the change.

---
In reply to: [1814391221](https://github.com/microsoft/onnxruntime/pull/19081#issuecomment-1814391221) [](http://example.com/codeflow?ancestors=1814391221)

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp [hunk 2]
```diff
@@ -559,8 +559,8 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
                 ML_CHECK_VALID_ARGUMENT(axis < outputShapeDimCount);
                 uint32_t broadcastAxisLength = outputShape[axis];
                 ML_CHECK_VALID_ARGUMENT((inputTensorShape[0] == broadcastAxisLength) ||
```
Comment (fdwr) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp:561]:
```suggestion
                ML_CHECK_VALID_ARGUMENT(
                    (inputTensorShape[0] == broadcastAxisLength) ||
                    // Treat as broadcast dimension to match CPU behavior.
                    (inputTensorShape[0] == 1)
                );
```

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp [hunk 3]
```diff
@@ -558,7 +558,13 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
...
                 ML_CHECK_VALID_ARGUMENT(axis < outputShapeDimCount);
                 uint32_t broadcastAxisLength = outputShape[axis];
-                ML_CHECK_VALID_ARGUMENT(inputTensorShape[0] == broadcastAxisLength);
+                ML_CHECK_VALID_ARGUMENT(
+                    (inputTensorShape[0] == broadcastAxisLength) ||
+                    // Treat as broadcast dimension to match CPU behavior.
+                    (inputTensorShape[0] == 1)
+                );
+                    // treat as broadcast dimension to match CPU behavior
+                    (inputTensorShape[0] == 1));
```
Comment (fdwr) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp:567]:
```suggestion
```

**Review by fdwr — APPROVED** [review]

**Review by fdwr — APPROVED** [review]

**Review by jeffbloo — APPROVED** [review]

**Pull request merged**